draft-ietf-acme-subdomains.md

```diff
@@ -279,7 +283,9 @@ In the following example newOrder payload, the client requests a certificate for
 
 If the client is unable to fulfill authorizations against a parent domain, the client should not include the "parentDomain" field.
 
-Server newOrder handling generally follows the process documented in ACME, {{Section 7.4 of RFC8555}}. If the server is willing to allow subdomain authorizations for the domain specified in "parentDomain", then it creates an authorization object against that parent domain and includes the "subdomainAuthAllowed" flag with a value of true. If the server policy does not allow creation of subdomain authorizations against that parent domain, then it can create an authorization object for the indicated identifier value, and includes the "subdomainAuthAllowed" flag with value of false.
+Server newOrder handling generally follows the process documented in ACME, {{Section 7.4 of RFC8555}}. If the server is willing to allow subdomain authorizations for the domain specified in "parentDomain", then it creates an authorization object against that parent domain and includes the "subdomainAuthAllowed" flag with a value of true.
+
+If the server policy does not allow creation of subdomain authorizations against that parent domain, then it can create an authorization object for the indicated identifier value, and SHOULD NOT include the "subdomainAuthAllowed" flag. As the client requested a subdomain authorization for the parent domain, and not for the indiciated identifier, there is no need for the server to include the "subdomainAuthAllowed" flag in the authorization object for the indicated identifier.
```

Comment:
If the server does not say false, then how can the client distinguish between a server that does not support acme-subdomains from one that has a policy against it?  A policy against it might be solved via administrators, or perhaps by purchasing the right service.
I think that it's worth saying something here.

Reply:
https://github.com/upros/acme-subdomains/blob/master/draft-ietf-acme-subdomains.md#authorization-object section already says:

> If the "subdomainAuthAllowed" field is not included, then the assumed default value is false.

is that sufficient?

Reply:
If a server has its metata indicate subdomainAuthhAllowed=true and client sends newAuthz for a domain with subdomainAuthhAllowed=true, but (i) gets back an authz without subdomainAuthhAllowed OR (ii) with subdomainAuthhAllowed=false, does it matter which of (i) or (ii) it is? Either way, client knows that subdomainAuthhAllowed=true at the server level, but it failed to get a subdomain auth for the given identifier. Same holds true for the ancestorDomain in the newOrder obect, doesnt it?

Reply:
> If a server has its metadata indicate subdomainAuthhAllowed=true and client sends newAuthz for a domain with subdomainAuthhAllowed=true, but (i) gets back an authz without subdomainAuthhAllowed OR (ii) with subdomainAuthhAllowed=false, does it matter which of (i) or (ii) it is? Either way, client knows that subdomainAuthhAllowed=true at the server level, but it failed to get a subdomain auth for the given identifier. Same holds true for the ancestorDomain in the newOrder obect, doesnt it?

Yes, I think that it does matter.
If it comes back without subdomainAuthhAllowed, then the server does not support it. (That's not negotiable at layer 8/9)
If it comes back with subdomainAuthhAllowed=false, then the server does support it, but has refused. That is negotiable.

Reply:
@mcr  So, if the server supports subdomainAuthAllowed (and indicates this in its metadata), then should it always include the subdomainAuthAllowed flag in all authorization objects, even if the client did not request a subdomain authz?

i.e. if client sends newAuthz without including the subdomainAuthhAllowed field, or client sends newOrder without including the ancestorDomain field, the server should always include the subdomainAuthAllowed field in the returned authz object. And whether the value is true or false depends on server policy,

And if so, is this a SHOULD or a MUST ?

Reply:
If a server has its metadata indicate subdomainAuthhAllowed=true, then the client knows that the server supports subdomains.

And if the client knows this, then why not allow ommission of subdomainAuthhAllowed in Authz objects with an implicit default value of false if absent? Just like for * authz, its an implicit default of false if absent.

Reply:
I defer to you, I might not understand all of the flow.